src/inmanta/server/services/compilerservice.py

```diff
@@ -500,7 +504,7 @@ class CompilerService(ServerSlice, environmentservice.EnvironmentListener):
 
     def __init__(self) -> None:
         super().__init__(SLICE_COMPILER)
-        self._recompiles: dict[uuid.UUID, Task] = {}
+        self._recompiles: dict[uuid.UUID, Task[Type[None]]] = {}
```

Comment:
Why `type[None]` instead of just `None`?

Reply:
I None valid? because that is the instance and not the type. But maybe that is no longer required.

Reply:
You write it all the time in function return types. Afaik `type[None]` is the type for `type(None)` (the type), while these tasks just return `None` (the instance).

Reply:
fixed